packages/Toast/src/Toast.js

```diff
@@ -73,77 +71,125 @@ const icons = {
 
 function Toast(props) {
   const {
-    ariaAlert,
-    canAutoClose,
     autoCloseDelay,
+    canAutoClose,
     children,
     hasCloseButton,
-    isOpen,
-    onClose,
     isFixed,
+    isOpen,
+    isPolite,
     kind,
+    onClose,
     zIndex,
     ...moreProps
   } = props;
 
-  const ariaLive = ariaAlert ? "assertive" : "polite";
   const [isToastOpen, setIsToastOpen] = React.useState(isOpen === undefined ? true : isOpen);
-  const timerRef = React.useRef(null);
-  const defaultZIndex = isFixed ? zIndexPlaceholder : null;
-
-  const memoizedStartTimer = React.useCallback(() => {
+  const [shouldRender, setShouldRender] = React.useState(!isPolite);
+  let autoCloseTimer = React.useRef(null).current;
+  let renderTimer = React.useRef(null).current;
+  const ariaRole = isPolite ? "status" : "alert";
+  const defaultZIndex = isFixed ? zValue(7) : null;
+  const isVisuallyHidden = kind === Kinds.VISUALLY_HIDDEN;
+
+  const memoizedStartAutoCloseTimer = React.useCallback(() => {
     function handleDelayedClose() {
-      clearTimeout(timerRef.current);
+      clearTimeout(autoCloseTimer);
       if (isOpen === undefined) setIsToastOpen(false);
       onClose();
     }
 
-    timerRef.current = setTimeout(handleDelayedClose, Math.max(autoCloseDelay, minimumCloseTimeout));
+    autoCloseTimer = setTimeout(handleDelayedClose, Math.max(autoCloseDelay, minimumCloseTimeout));
```

Comment:
why this change? This does not work. It's just assigning to a local variable, which will be discarded after the current render. You do need to assign to `.current`.

Reply:
🤦‍♂